consumer/src/main/kotlin/streams/procedures/StreamsSinkProcedures.kt

```diff
@@ -27,13 +39,31 @@ class StreamsSinkProcedures {
         }
 
         val properties = config?.mapValues { it.value.toString() } ?: emptyMap()
-        val configuration = StreamsSinkProcedures.streamsEventSinkConfigMapper.convert(config = properties)
+        val configuration = streamsEventSinkConfigMapper.convert(config = properties)
 
         val data = readData(topic, config ?: emptyMap(), configuration)
 
         return data.map { StreamResult(mapOf("data" to it)) }.stream()
     }
 
+    @Procedure("streams.sink.start")
+    fun sinkStart(): Unit {
+        checkEnabled()
+        streamsEventSink?.start()
+    }
+
+    @Procedure("streams.sink.stop")
```

Comment:
This should return a boolean to indicate success/failure.   You can return true ("yes, I stopped it") if it never began, or wasn't configured.  Because the user asked to stop, and that succeeded in those cases.

Reply:
done